React/Modules/RCTInterpolationAnimatedNode.m

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "RCTInterpolationAnimatedNode.h"
+#import "RCTAnimation.h"
+
+@interface RCTInterpolationAnimatedNode ()
+
+@property (nonatomic, weak) RCTValueAnimatedNode *parentNode;
+@property (nonatomic, strong) NSArray *inputRange;
+@property (nonatomic, strong) NSArray *outputRange;
+
+@end
+
+@implementation RCTInterpolationAnimatedNode
+
+- (instancetype)initWithTag:(NSNumber *)tag config:(NSDictionary *)config {
+  self = [super initWithTag:tag config:config];
+  if (self) {
+    self.inputRange = config[@"inputRange"];
+    NSMutableArray *outputRange = [NSMutableArray array];
+    for (id value in config[@"outputRange"]) {
+      if ([value isKindOfClass:[NSNumber class]]) {
+        [outputRange addObject:value];
+      }
+      if ([value isKindOfClass:[NSString class]]) {
+        NSString *str = (NSString *)value;
+        if ([str containsString:@"deg"]) {
+          NSString *newString = [str stringByReplacingOccurrencesOfString:@"deg" withString:@""];
+          double degrees = newString.floatValue;
+          [outputRange addObject:@(DegreesToRadians(degrees))];
+        } else if ([str containsString:@"rad"]) {
+          NSString *newString = [str stringByReplacingOccurrencesOfString:@"rad" withString:@""];
+          [outputRange addObject:@(newString.floatValue)];
+        }
+        
+      }
+    }
+    
+    self.outputRange = outputRange;
+  }
+  return self;
+}
+
+- (void)onAttachedToNode:(RCTAnimatedNode *)parent {
+  [super onAttachedToNode:parent];
+  if ([parent isKindOfClass:[RCTValueAnimatedNode class]]) {
+    self.parentNode = (RCTValueAnimatedNode *)parent;
+  }
+  
+}
+
+- (void)onDettachedFromNode:(RCTAnimatedNode *)parent {
+  [super onDettachedFromNode:parent];
+  if (self.parentNode == parent) {
+    self.parentNode = nil;
+  }
+}
+
+- (NSUInteger)findRangeIndexForValue:(NSNumber *)value {
```

Comment:
Can you document what this does? The name is a bit confusing because it talks about a range but then it returns a single integer instead of a range of values.